client/lib/media/store.js

```diff
@@ -31,6 +31,15 @@ function receiveSingle( siteId, item, itemId ) {
 		}
 
 		MediaStore._pointers[ siteId ][ itemId ] = item.ID;
+
+		const maybeTransientMediaItem = MediaStore._media[ siteId ][ itemId ];
+
+		if ( maybeTransientMediaItem && maybeTransientMediaItem.transient ) {
```

Comment:
Woot, I even forgot that I added this function a while back. Thanks for the hint!